docs/source/rule_generation.rst

```diff
@@ -0,0 +1,65 @@
+++++++++++++++++++++++++++++++++++++++++
+Rule Generation
+++++++++++++++++++++++++++++++++++++++++
+
+The Rule generation technique is based on the folowing idea belows:
+
+1. Sort API by API usage count
+2. Separate all API into two groups, P(20% least usage count) and S(other 80% API), by the Pareto principle (20-80 rule)
+3. Combine $P$ and $S$ into four different phases:
+    - PxP
+    - PxS 
+    - SxP 
+    - SxS
+4. Execute rule generate by phase: PxP -> PxS -> SxP -> SxS
```

Comment:
I suggest changing this sentence to "Execute the rule generation with each phase in this order: ...". In this way, it becomes more clear IMHO.